commons/src/test/java/org/eclipse/kapua/commons/setting/KapuaSettingExceptionTest.java

```diff
@@ -0,0 +1,83 @@
+
+    Throwable throwable = new Throwable("throwable_error_message");
+
+    @Test
+    public void constructorNullTest() {
+        KapuaSettingException kapuaSettingException = new KapuaSettingException(null);
+        assertNull("Null expected!", kapuaSettingException.getCode());
+    }
+
+    @Test
+    public void constructorInvalidResourceNameTest() {
+        KapuaSettingException kapuaSettingException = new KapuaSettingException(KapuaSettingErrorCodes.INVALID_RESOURCE_NAME);
+        assertEquals("Expected and actual values should be the same!", "INVALID_RESOURCE_NAME", kapuaSettingException.getCode().toString());
+    }
+
+    @Test
+    public void constructorInvalidResourceFileTest() {
+        KapuaSettingException kapuaSettingException = new KapuaSettingException(KapuaSettingErrorCodes.INVALID_RESOURCE_FILE);
+        assertEquals("Expected and actual values should be the same!", "INVALID_RESOURCE_FILE", kapuaSettingException.getCode().toString());
+    }
+
+    @Test
+    public void constructorResourceNotFoundTest() {
+        KapuaSettingException kapuaSettingException = new KapuaSettingException(KapuaSettingErrorCodes.RESOURCE_NOT_FOUND);
+        assertEquals("Expected and actual values should be the same!", "RESOURCE_NOT_FOUND", kapuaSettingException.getCode().toString());
+    }
+
+    @Test
+    public void extendedConstructorWithObjectArgTest() {
+        Object[] arguments = new Object[]{"invalid_resource_name", 1, 10L, 10.34, 5.99f, 'D', "", true, false, -128, 127, -32768, 32767, -2147483648, 2147483647, -9223372036854775808L, 9223372036854775807L};
+        for (Object value : arguments) {
+            KapuaSettingException kapuaSettingException = new KapuaSettingException(KapuaSettingErrorCodes.INVALID_RESOURCE_NAME, value);
+            assertTrue("Expected and actual values should be the same!", kapuaSettingException.getCode().toString().equals("INVALID_RESOURCE_NAME") && kapuaSettingException.getMessage().equals("Error: " + value));
+        }
+    }
+
+    @Test
+    public void extendedConstructorWithNullObjArgTest() {
+        KapuaSettingException kapuaSettingException = new KapuaSettingException(KapuaSettingErrorCodes.INVALID_RESOURCE_NAME, (Object) null);
+        assertTrue("Expected and actual values should be the same!", kapuaSettingException.getCode().toString().equals("INVALID_RESOURCE_NAME") && kapuaSettingException.getMessage().equals("Error: " + null));
+    }
+
+    @Test
+    public void extendedConstructorWithThrowableAndNullObjArgTest() {
+        KapuaSettingException kapuaSettingException = new KapuaSettingException(KapuaSettingErrorCodes.INVALID_RESOURCE_NAME, throwable, (Object) null);
+        assertTrue("Expected and actual values should be the same!", kapuaSettingException.getCode().toString().equals("INVALID_RESOURCE_NAME") && kapuaSettingException.getCause().equals(throwable) && kapuaSettingException.getMessage().equals("Error: " + null));
```

Comment:
Please split this into 2 assertEquals otherwise the real cause of the assertFailure will be hidden and uncertain.